### ISSUE TITLE
Version 2.6.0 and 2.6.0.1 not allowing WiFi

### ISSUE THREAD

**Issue by Gitoffthelawn** [ukanth/afwall#591]
Upgraded from v2.5.2 to v2.6.0 (F-Droid) on a KitKat device.  WiFi and cell data working as expected with v2.5.2.  With v2.6.0, cell data works perfectly, but WiFi gets no response.

Before I spend hours trying to figure it out (already spent 1 hour!), any idea why this would happen?  No other changes made beside upgrading F-Droid to v2.6.0 and rebooting.

**Comment by ukanth** [comment]
Please post iptables rules.

**Comment by vestingz** [comment]
Can confirm. Just reverted to 2.5.2, rewrote iptables rules and it worked again. 
Edit: Network Log showed outgoing packets, but no incoming.

**Comment by Gitoffthelawn** [comment]
@vestingz Thank you so much for confirming.  That's exceptionally helpful.

Did you revert to 2.5.2 via F-Droid?

When I tried to do that, it issued a severe warning about potential data loss, and given that AFWall+ is a root app, I didn't want to mess anything up.  Any troubles with reverting?

@ukanth Can you imagine any troubles with reverting from 2.6.0 to 2.5.2 via F-Droid?

**Comment by ukanth** [comment]
It should not, I need logcat/iptables rules to look at

**Comment by Gitoffthelawn** [comment]
@ukanth Sounds good.  I just emailed you the files.  Pw is First_Last (your info).

**Comment by vestingz** [comment]
@Gitoffthelawn No, neither WLAN nor mobile worked so i downloaded the apk to my lap and manually installed 2.5.2 after uninstalling 2.6.0.

**Comment by Gitoffthelawn** [comment]
@vestingz Thanks.  When you say "neither WLAN nor mobile worked", do you mean after downgrading using F-Droid?

Also, do you have any logs/iptables you can give to @ukanth?

**Comment by vestingz** [comment]
@Gitoffthelawn No, before. sorry for being unclear. It seemed like the iptables of 2.6.0 allowed outbound traffic only on both adapters as Network Log recorded. Unfortunately i didn't save any logs, sorry.

**Comment by ukanth** [comment]
@Gitoffthelawn , I have not received any report. Please send it again. Also try to catch this output as well

iptables -S ( in console) and send it to me.

**Comment by Gitoffthelawn** [comment]
@ukanth OK... I'll send it again in a few minutes.

**Comment by Gitoffthelawn** [comment]
@ukanth Sent.  Let me know if you got it.  I didn't have time to get iptables -S, so if you still need that, let me know, and I'll get it for you.

**Comment by ukanth** [comment]
@Gitoffthelawn , May I know to which email id you are sending these emails. Because I'm not getting those.

**Comment by Gitoffthelawn** [comment]
Yes, you may. :)  I sent the first one to cumakt+afwall@gmail.com and the second to cumakt+github@gmail.com, so they should have both gone to cumakt@gmail.com

**Comment by Gitoffthelawn** [comment]
Try this: http://dropjar.com/#1g2ckqny

**Comment by Gitoffthelawn** [comment]
Let me know if you got it.  Thanks.  In 3 minutes, I will be AFK for at least an hour, possibly longer...

**Comment by ukanth** [comment]
Nope, not by email, but got it via the above link. it seems password protected.

**Comment by Gitoffthelawn** [comment]
@ukanth Weird how emails are not getting through, but glad you got the file.  To view it, see https://github.com/ukanth/afwall/issues/591#issuecomment-251578518

**Comment by Gitoffthelawn** [comment]
Spent a couple hours performing some experiments.  Here's a summary of what I learned:
- Cannot use F-Droid to downgrade from v2.6.0 to v2.5.2: it fails every time
- AFWall+ v2.6.0.1 (manually installed from apk) had the same problems with WiFi as v2.6.0
- Manually reverting to v2.5.2 fixed all the WiFi issues (using the same settings as with v2.6.0 and v2.6.0.1)

**Comment by ukanth** [comment]
@Gitoffthelawn, Unfortunately I can't open the zip with the hint you provided. 

Also I tried reverting the change which might have caused this issue, purely based on my guess. If you can test this, would be great.

@vestingz ..

https://www.dropbox.com/s/rm6v7xurvna63fw/AFWall%2B%20TEST%202.7.0.apk?dl=0

**Comment by Gitoffthelawn** [comment]
@ukanth It's your first name, then an underscore, followed by your last name.  Your first name and last name each have the first letter capitalized.  I'll owe you an apology if I accidentally spelled your name wrong! :smile:

**Comment by ukanth** [comment]
@Gitoffthelawn , looks like you have spelled it wrong or I have some other firstname/lastname. I tried all combinations !

**Comment by Gitoffthelawn** [comment]
@ukanth LOL... You have my apology!

I used `Umakanthan_Chandran`

Is the about page wrong?

**Comment by ukanth** [comment]
@Gitoffthelawn , Since you have uploaded the service which encrypts the zip, the password doesn't seems to work.

**Comment by Gitoffthelawn** [comment]
@ukanth My apologies; the file definitely had an issue.  Try this: http://dropjar.com/#1mqaazi6

This issue was getting cluttered, so I cleaned up what I could.  You may desire to do the same.

**Comment by vestingz** [comment]
@ukanth Thanks for the link :) I will test within the next days.
@Gitoffthelawn What's wrong with posting your rules on pastebin or equal?

**Comment by Gitoffthelawn** [comment]
@vestingz Pastebin has been overloaded and was only showing this: http://pastebin.com/i/503.jpg

What's "equal"?  Do you have a url?

**Comment by vestingz** [comment]
@Gitoffthelawn paste.ee | 0bin.net | pastie.org | nopaste.me just to name a few.

**Comment by aligitor** [comment]
I have similar problem, I am not sure if it's the same or not. Have an UMI Zero phone with KitKat, I flashed a custom CyanogenMod rom on it (Lollypop), but I had problems with connecting to wifi networks. I used the newest stable release of afwall+ on it. I thought it's a bug in the rom, and after one month I flashed the original official rom (a fresh new install, not a backup file). Had the same issue, even though I tried a dozen of roms and configuration on it. I gave up, thinking that it must be a hardware problem.

Now I got a Huawei P8 Lite phone (Android 6 Marshmallow). The wifi worked fine, even after unlocking the bootloader and rooting the phone. However after I installed the newest afwall+ from F-Droid, the wifi stopped working, showing the same symptoms as my UMI.

Disabling afwall, downgrading or removing didn't help. I had a full backup of the working state of the system (with TWRP), restoring the backup didn't help either. I also tried to remove all iptable rules from terminal with the commands I found here: https://www.digitalocean.com/community/tutorials/how-to-list-and-delete-iptables-firewall-rules (The 'Flush All Rules, Delete All Chains, and Accept All' section).

Mobile data works fine.

Is there anything afwall+ may changed in the core system settings which could "survive" after factory reset or installing new roms?

I heard that some system apps may try to ping google servers, and if they can't, some wifi networks may be blocked as it doesn't provide internet connection. I used afwall in "allow selected" mode, system processes were blocked too. (However enabling them didn't solve my problem.)

I also try to use my Android without Google Account: Play Store, Youtube, GMail, Drive and other GApps are frozen with Link2sd. (I used my UMI for a year this way, and didn't have problem with wifi.)

If you think that my problem is related to this issue, feel free to ask for logs or anything. I also appreciate any help or suggestion.
